snuba/web/bulk_delete_query.py

```diff
@@ -0,0 +1,183 @@
+from snuba.utils.metrics.util import with_span
+from snuba.utils.metrics.wrapper import MetricsWrapper
+from snuba.utils.schemas import ColumnValidator, InvalidColumnType
+from snuba.utils.streams.configuration_builder import build_kafka_producer_configuration
+from snuba.utils.streams.topics import Topic
+from snuba.web.delete_query import (
+    DeletesNotEnabledError,
+    _construct_condition,
+    _enforce_max_rows,
+    _get_attribution_info,
+    deletes_are_enabled,
+)
+
+metrics = MetricsWrapper(environment.metrics, "snuba.delete")
+logger = logging.getLogger(__name__)
+
+ConditionsType = Mapping[str, Sequence[str | int | float]]
+
+
+class DeleteQueryMessage(TypedDict):
+    rows_to_delete: int
+    storage_name: str
+    conditions: ConditionsType
+    tenant_ids: Mapping[str, str | int]
+
+
+delete_kfk: Producer | None = None
+
+
+def _delete_kafka_producer() -> Producer:
+    global delete_kfk
+    if delete_kfk is None:
+        delete_kfk = Producer(
+            build_kafka_producer_configuration(
+                topic=Topic.LW_DELETIONS,
+            )
+        )
+    return delete_kfk
+
+
+def _record_query_delivery_callback(
+    error: Optional[KafkaError], message: KafkaMessage
+) -> None:
+    metrics.increment(
+        "record_query.delivery_callback",
+        tags={"status": "success" if error is None else "failure"},
+    )
+
+    if error is not None:
+        logger.warning("Could not produce delete query due to error: %r", error)
+
+
+def produce_delete_query(delete_query: DeleteQueryMessage) -> None:
+    try:
+        producer = _delete_kafka_producer()
+        data = rapidjson.dumps(delete_query)
+        producer.poll(0)  # trigger queued delivery callbacks
+        producer.produce(
+            settings.KAFKA_TOPIC_MAP.get(
+                Topic.LW_DELETIONS.value, Topic.LW_DELETIONS.value
+            ),
+            data.encode("utf-8"),
+            on_delivery=_record_query_delivery_callback,
+            # TODO(should we add headers? or key)
+        )
+    except Exception as ex:
+        logger.exception("Could not produce delete query due to error: %r", ex)
+
+
+@with_span()
+def delete_from_storage(
+    storage: WritableTableStorage,
+    conditions: Dict[str, list[Any]],
+    attribution_info: Mapping[str, Any],
+) -> dict[str, Result]:
+    """
+    Copied over from delete_query.py with some modifications
+    """
+    if not deletes_are_enabled():
+        raise DeletesNotEnabledError("Deletes not enabled in this region")
+
+    delete_settings = storage.get_deletion_settings()
+    if not delete_settings.is_enabled:
+        raise DeletesNotEnabledError(
+            f"Deletes not enabled for {storage.get_storage_key().value}"
+        )
+
+    if list(conditions.keys()).sort() != list(delete_settings.allowed_columns).sort():
+        raise InvalidQueryException(
+            f"Invalid Columns to filter by, must be in {delete_settings.allowed_columns}"
+        )
```

Comment:
To avoid sorting lists explicitly, can we do this?

```
difference =  set(conditions.keys()) - set(delete_settings.allowed_columns)
if difference != set():
    raise InvalidQueryException(
        f"Invalid Columns {difference} to filter by, must be in {delete_settings.allowed_columns}"
    )
```